tests/test_schema.py

```diff
@@ -0,0 +1,221 @@
+"""
+
+Tests for schema and examples.
+
+"""
+
+import json
+import glob
+from pathlib import Path, PurePath
+import datetime
+
+import pytest
+import yaml
+
+import jsonschema
+
+from copy import deepcopy
+
+
+TESTDIR = Path(__file__).parent.absolute()
+
+
+def test_schema_basic_json_syntax():
+    """Confirm that schemas are valid JSON"""
+
+    # find and parse all schema files
+    path = PurePath(TESTDIR, "../schema/definitions/*/schema/*.json")
+    schema_file_paths = glob.glob(str(path))
+
+    # check that schemas are there
+    assert len(schema_file_paths) > 0
+
+    for schema_file_path in schema_file_paths:
+        _parse_json(schema_file_path)
+
+
+def test_schema_example_filenames():
+    """Assert that all examples are .yml, not .yaml"""
+
+    searchpath = str(PurePath(TESTDIR, "../schema/definitions/*/examples/*.*"))
+    filenames = glob.glob(searchpath)
+
+    # check that examples are there
+    assert len(filenames) > 0
+
+    for filename in filenames:
+        assert filename.endswith(".yml"), filename
+
+
+def test_schema_080_validate_examples_as_is():
+    """Confirm that examples are valid against the schema"""
+
+    # parse the schema
+    schema = _parse_json(
+        str(PurePath(TESTDIR, "../schema/definitions/0.8.0/schema/fmu_results.json"))
+    )
+    examples = [
+        _parse_yaml(str(path))
+        for path in glob.glob(
+            str(PurePath(TESTDIR, "../schema/definitions/0.8.0/examples/*.yml"))
+        )
+    ]
+
+    for example in examples:
+        jsonschema.validate(instance=example, schema=schema)
+
+
+def test_schema_080_logic_case():
+    """Asserting validation failure when illegal contents in case example"""
+
+    # parse the schema and one example
+    schema = _parse_json(
+        str(PurePath(TESTDIR, "../schema/definitions/0.8.0/schema/fmu_results.json"))
+    )
+    example = _parse_yaml(
+        str(PurePath(TESTDIR, "../schema/definitions/0.8.0/examples/case.yml"))
+    )
+
+    # assert validation with no changes
+    jsonschema.validate(instance=example, schema=schema)
+
+    # assert validation error when "fmu" is missing
+    _example = deepcopy(example)
+    del _example["fmu"]
+
+    with pytest.raises(jsonschema.exceptions.ValidationError):
+        jsonschema.validate(instance=_example, schema=schema)
+
+    # assert validation error when "fmu.model" is missing
+    _example = deepcopy(example)
+    del _example["fmu"]["model"]
+
+    with pytest.raises(jsonschema.exceptions.ValidationError):
+        jsonschema.validate(instance=_example, schema=schema)
+
+
+def test_schema_080_logic_field_outline():
+    """
+    Test content-specific rule:
+        When content == field_outline, require the field_outline field
```

Comment:
Ideally all docstrings shall be like this:
```
"""A single line as a summary, ending with punctuation.

Then, after space, more description...
"""
```

Reply:
✔